client/react/lib/index.tsx

```diff
@@ -29,75 +29,126 @@ interface setComponent {
   return <FallbackComponent dynamicoStatus={status} {...props} />;
-}
+};
 
 export const dynamico = function<T = any>(
   name: string,
   { fallback = null, devMode = false, ...options }: ComponentOptions<T> = {}
 ): FunctionComponent<T> {
   return (props: T) => {
     const [Component, setComponent]: [Component, setComponent] = useState({});
-    const [status, setStatus]: [Status, setStatus] = useState<Status>({currentStatus: ComponentStatus.Loading});
-    
-    const dynamicoClient = useContext<DynamicoClient | undefined>(DynamicoContext);
-    let release = () => {};
-
-    const getComponent = async () => {
-      if (!dynamicoClient) {
-        throw `Couldn't find dynamico client in the context, make sure you use DynamicoContext.Provider`;
-      }
-      if (devMode) {
-        const devClient = new DynamicoDevClient({
-          dependencies: dynamicoClient.dependencies,
-          ...(typeof devMode === 'object' ? devMode : {}),
-          callback: (view: any) => setComponent({ view })
-        });
-
-        release = await devClient.get(name, options);
-
-        return;
-      }
-      
-      setComponent({ view: await dynamicoClient.get(name, options) });
-    };
+    const [status, setStatus]: [Status, setStatus] = useState<Status>({ currentStatus: ComponentStatus.Loading });
+
+    const { client: dynamicoClient, devClient } = useContext(DynamicoContext);
 
     useEffect(() => {
-      
-        getComponent().catch((error) => {
-          setStatus({
-            currentStatus: ComponentStatus.Error,
-            error
+      let release = () => {};
+
+      const setError = (error: Error) => setStatus({ currentStatus: ComponentStatus.Error, error });
+
+      const getComponent = async () => {
+        if (!dynamicoClient) {
+          throw `Couldn't find dynamico client in the context, make sure you use DynamicoContext.Provider`;
+        }
+
+        if (devMode || devClient) {
+          const devOptions = typeof devMode === 'object' ? devMode : {};
+
+          const client =
+            devClient ||
+            new DynamicoDevClient({
+              dependencies: dynamicoClient.dependencies,
+              ...devOptions
+            });
+
+          let usingFallbackComponent = false;
+
+          release = client.get(name, {
+            interval: devOptions.interval,
+            ...options,
+            callback: async (err, view: any) => {
+              if (!err) {
+                usingFallbackComponent = false;
+                return setComponent({ view });
+              }
+
+              console.warn(`failed getting component ${name} from dev server`, err);
+
+              if (!devMode) {
+                return setError(err);
+              }
+
+              if (usingFallbackComponent) {
+                return;
+              }
+              usingFallbackComponent = true;
+
+              try {
+                setComponent({ view: await dynamicoClient.get(name, options) });
```

Comment:
Is this the fallback for a devClient that failed to get a component?
Shouldn't we use the actual `<FallbackBuilder...` here? If I understand correctly this creates a behavior that says that if a dev server failed to provide a component I'm going to the remote registry, so I'll be getting an older version of the component? Is this the correct behavior here? Am I missing something?

Reply:
I thought that's what we talked about. If we're using workspaces and failed to get a component from local dev server then try the remote server. In case not all components are stored locally

Reply:
Also, if you add a new component while being in global dev mode it won't show anything if there's a problem with the dev server... or you'll render undefined... no way for you to test the fallback component property you pass in...

Reply:
so maybe we should have the fallback as well here? just in case this is a new comp? So the strategy is dev server > remote registry > local fallback

wdyt?

Reply:
wait... this is the behavior now? right?

Reply:
all I'm worried about is if you can test your local fallback in case the component doesn't exist on the server... if it's possible than forget about this comment...

Reply:
if the remote server returns 404 for non existing components, it will render falback